protocol/auditor.go

```diff
@@ -0,0 +1,176 @@
+// This module implements a generic CONIKS auditor, i.e. the
+// functionality that clients and auditors need to verify
+// a server's STR history.
+
+package protocol
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+)
+
+// Auditor provides a generic interface allowing different
+// auditor types to implement specific auditing functionality.
+type Auditor interface {
```

Comment:
Also, thinking ahead to implementing a CoSi-compatible auditor, AFAICT, it should be able to directly use `AudState` (or whatever we call it) and implement `AuditDirectory()` like we've done so far.